cluster/cluster_test.go

```diff
@@ -334,7 +334,7 @@ func testTLSConnection(t *testing.T) {
 	)
 	require.NoError(t, err)
 	go p2.Settle(context.Background(), 0*time.Second)
-
+	p2.WaitReady(context.Background())
```

Comment:
Drive-by fix; this test flaked a couple times. https://github.com/prometheus/alertmanager/issues/3593

The `go Settle` and following assert race each other. Other tests, including further up in this same test, use a `WaitReady` - it seems to be missing here.